operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,20 @@ type AWSCSIDriverConfigSpec struct {
 	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b|aws-iso-e|aws-iso-f):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetricsRefreshPeriod specifies the frequency, in minutes, at which volume metrics are refreshed.
+	// This field enables metric collection when set to a value greater than 0.
+	// The default refresh period in the EFS CSI driver is 240 minutes.
+	// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+	// +optional
+	EFSVolumeMetricsRefreshPeriod int `json:"efsVolumeMetricsRefreshPeriod,omitempty"`
+
+	// efsVolumeMetricsFSRateLimit defines the rate limit, in goroutines per file system, for processing volume metrics.
+	// Requires efsVolumeMetricsRefreshPeriod to be set to a value greater than 0.
```

Comment:
Creating combinations like this typically results in a discriminated union, rather than creating paired fields.

```
efsVolume:
  metrics:
    state: Enabled
    enabled:
      refreshPeriod: ...
      rateLimit: ...
```

Reply:
Do you have an example? Is `enabled` a struct? How to make it mandatory when `state` is `Enabled`?

Reply:
Reworked the PR, PTAL.

Reply:
Example union in case you haven't found one, https://github.com/openshift/api/blob/ba11c1587003dc84cb014fd8db3fa597a3faaa63/example/v1/types_stable.go#L108-L125